### PR TITLE
test(dtslint): add dtslint test for iif observable static function (#…

### PR DESCRIPTION
**Pull request by dkosasih**
**Description:** Add dtslint test for `iif` observable static funtion

**Related issue (if exists):** #4093

### CODE REVIEW

File: spec-dtslint/observables/iif-spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { iif, of } from 'rxjs';
+
+it('should accept function as first parameter', () => {
+  // Typescript limitation: cannot infer default value in params
+  // e.g.: const a = iif(() => false, EMPTY, EMPTY) // $ExpectType Observable<never>
+  // However, test below does not infer Observable<never> if no params is passed
+  // although the default value for second and third param is EMPTY observable
+  const a = iif(() => false); // $ExpectType Observable<{}>
```
Comment (benlesh) [inline spec-dtslint/observables/iif-spec.ts]: This seems like a bug... we could add a type overload to cover this.

File: spec-dtslint/observables/iif-spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+});
+
+it('should infer correctly with 2 parameters', () => {
+  const a = iif(() => false, of(1)); // $ExpectType Observable<number | {}>
```
Comment (benlesh) [inline spec-dtslint/observables/iif-spec.ts]: should be `Observable<number | never>` ... which I think results in `Observable<number>` regardless?

Add a type overload to account for this.

File: spec-dtslint/observables/iif-spec.ts (hunk 1 above)
Comment (dkosasih) [reply spec-dtslint/observables/iif-spec.ts]: The comments should be able to remove when #4494 is merged.

**Comment by cartant** [comment]
Closed in favour of #5176 - which cherry-picks this PR's commit.

**Pull request closed without merging**